pkg/scheduler/objects/queue.go

```diff
@@ -1609,20 +1612,38 @@ func (sq *Queue) updateMaxResourceMetrics() {
 
 // updateAllocatedResourceMetrics updates allocated resource metrics for all queue types.
 func (sq *Queue) updateAllocatedResourceMetrics() {
+	// Special case when we prune the allocated resources to zero, we need to update the metrics
+	// here only keep basic types memory and vcores to make sure we have updated metrics
+	if sq.allocatedResource.IsEmpty() {
+		metrics.GetQueueMetrics(sq.QueuePath).SetQueueAllocatedResourceMetrics("memory", float64(0))
+		metrics.GetQueueMetrics(sq.QueuePath).SetQueueAllocatedResourceMetrics("vcores", float64(0))
+	}
 	for k, v := range sq.allocatedResource.Resources {
 		metrics.GetQueueMetrics(sq.QueuePath).SetQueueAllocatedResourceMetrics(k, float64(v))
 	}
 }
 
 // updatePendingResourceMetrics updates pending resource metrics for all queue types.
 func (sq *Queue) updatePendingResourceMetrics() {
+	// Special case when we prune the pending resources to zero, we need to update the metrics
+	// here only keep basic types memory and vcores to make sure we have updated metrics
+	if sq.pending.IsEmpty() {
+		metrics.GetQueueMetrics(sq.QueuePath).SetQueuePendingResourceMetrics("memory", float64(0))
+		metrics.GetQueueMetrics(sq.QueuePath).SetQueuePendingResourceMetrics("vcores", float64(0))
+	}
```

Comment:
Same as for the allocated metrics update.

Reply:
Addressed in latest PR.